snippets/csharp/VS_Snippets_CLR/RegularExpressions.Examples.Email/cs/example4.cs

```diff
@@ -5,52 +5,48 @@
 
 public class RegexUtilities
 {
-   bool invalid = false;
+    public static bool IsValidEmail(string email)
+    {
+        if (string.IsNullOrEmpty(email))
+            return false;
 
-   public bool IsValidEmail(string strIn)
-   {
-       invalid = false;
-       if (String.IsNullOrEmpty(strIn))
-          return false;
+        // Use IdnMapping class to convert Unicode domain names.
+        try
+        {
+            // Examines the domain part of the email and normalizes it.
+            string DomainMapper(Match match)
```

Comment:
My own style has been to put any local functions at the end of the containing function, but that's not a standard.

Reply:
Great minds think alike.